packages/fcl/src/wallet-provider-spec/draft-v3.md

```diff
@@ -0,0 +1,494 @@
+
+  services: [                          // All the stuff that configures FCL
+    
+    // Authentication Service - REQUIRED
+    {
+      f_type: "Service",                                   // Its a service!
+      f_vsn: "1.0.0",                                      // Follows the v1.0.0 spec for the service
+      type: "authn",                                       // the type of service it is
+      method: "DATA",                                      // Its data!
+      uid: "amazing-wallet#authn",                         // A unique identifier for the service
+      endpoint: "your-url-that-fcl-will-render-in-iframe", // should be the same as was passed into the config
+      id: "0xUSER",                                        // the wallets internal id for the user, use flow address if you dont have one
+      // The Users Info
+      identity: {
+        f_type: "Identity",  // Its an Identity!
+        f_vsn: "1.0.0",      // Follows the v1.0.0 spec for an identity
+        address: "0xUSER",   // The users address
+        keyId: 0,            // OPTIONAL - The Users KeyId they will use
+      },
+      // The Wallets Info
+      provider: {
+        f_type: "ServiceProvider",      // Its a Service Provider
+        f_vsn: "1.0.0",                 // Follows the v1.0.0 spec for service providers
+        address: "0xWallet",            // A flow address owned by the wallet
+        name: "Amazing Wallet",         // OPTIONAL - The name of your wallet. ie: "Dapper Wallet" or "Blocto Wallet"
+        description: "The best wallet", // OPTIONAL - A short description for your wallet
+        icon: "https://___",            // OPTIONAL - Image url for your wallets icon
+        website: "https://___",         // OPTIONAL - Your wallets website
+        supportUrl: "https://___",      // OPTIONAL - An url the user can use to get support from you
+        supportEmail: "help@aw.com",    // OPTIONAL - An email the user can use to get support from you
+      },
+    },
+
+    // Authorization Service
+    {
+      f_type: "Service",
+      f_vsn: "1.0.0",
+      type: "authz",
+      uid: "amazing-wallet#authz",
+      // We will cover this at length in the authorization section of this guide
+    },
+    
+    // User Signature Service
+    {
+      f_type: "Service",
+      f_vsn: "1.0.0",
+      type: "user-signature",
+      uid: "amazing-wallet#user-signature",
+      // We will cover this at length in the user signature section of this guide
+    },
+
+    // OpenID Service
+    {
+      f_type: "Service",
+      f_vsn: "1.0.0",
+      type: "open-id",
+      uid: "amazing-wallet#open-id",
+      method: "DATA",
+      data: { // only include data that was request, ideally only if the user approves the sharing of data, everything is optional
+        f_type: "OpenID",
+        f_vsn: "1.0.0",
+        profile: {
+          name: "bob",
+          family_name: "builder", // icky underscored names because of OpenID Connect spec
```

Comment:
This stuff needs to be in a polling response now i think. @gregsantos should know more.